### PR TITLE
New doc: Order Form Sync Feature

### PR DESCRIPTION
**Pull request by mariana-caetano**
This PR adds the new documentation, **Implementing Order Form Sync feature**. This feature aims to maintain the user's session and order information across different pages, such as My Account and Checkout, in the FastStore environment.

### CODE REVIEW

**Review by beatrizmaselli — APPROVED** [review]
LGTM! 🚀

File: docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+---
+title: "Implementing Order Form Sync"
+sidebar_label: "Implementing Order Form Sync"
+---
+
+The Order Form Sync feature maintains the user's session and order information across different pages, such as My Account and Checkout, in the FastStore environment. 
+
+Pages like My Account, Login, and Checkout are in the Legacy Portal (vtexcommercestable) by default. Since FastStore is the main domain and pages in the legacy portal are subdomains, the user's session cookies for these domains must be synchronized to maintain a seamless shopping experience without losing progress or data.
+
+This synchronization is achieved by duplicating the session cookie for the main domain (faststore) whenever the user navigates to the portal (secure.{dominio}), and a service is called to update the Order Form information associated with that session. This ensures that the user's order data is synchronized across different pages.
```
Comment (eduardoformiga) [inline docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md]:
```suggestion
This synchronization is achieved by duplicating the session cookie for the main domain (faststore) whenever the user navigates to the portal (secure.<domain>), and a service is called to update the Order Form information associated with that session. This ensures that the user's order data is synchronized across different pages.
```

File: docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    });
+}
+
+callUrlWithRetry(`/api/io/faststore/cookie`, 3, 500)
+```
+4. Click on `Save`.
+
+:::caution 
+
+It is important to perform a retry as, the session cookie may not be immediately available in the request even after the script has run.
```
Comment (eduardoformiga) [inline docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md]:
```suggestion
It is important to perform a retry as the session cookie may not be immediately available in the request even after the script has run.
```

File: docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+:::
+
+### Step 2: Update FastStore Theme Code
+
+To send the `orderFormId` parameter, you must perform changes in the FastStore theme code. In your FastStore theme code, navigate to following files:
```
Comment (eduardoformiga) [inline docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md]:
```suggestion
To send the `orderFormId` parameter, you must perform changes in the FastStore starter code. Navigate to following files:
```

File: docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        +  href={
+        +    person?.id ? `/account?orderFormId=${id}` : +`/login?orderFormId=${id}`
+        +  }
+        className={`${styles.fsButton} text__title-mini`}
+        variant="tertiary"
+        icon={<Icon name="User" width={18} height={18} weight="bold" />}
+
+```
+#### `src/pages/account.tsx` 
+Responsible for rendering the My Account page. Add the following to allow subsequent pages to reference the the user’s session cookie of this page:
```
Comment (eduardoformiga) [inline docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md]:
```suggestion
Responsible for rendering the My Account page. Add the following to allow subsequent pages to reference the user’s session cookie on this page:
```

File: docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  useEffect(() => {
+  +  window.location.href = `${storeConfig.accountUrl}${window.location.search}`
+  }, [])
+
+  return (
+
+```
+
+#### `src/pages/login.tsx` 
+Responsible for rendering the Login page. Add the following to allow subsequent pages to reference the the user’s session cookie of this page:
```
Comment (eduardoformiga) [inline docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md]:
```suggestion
Responsible for rendering the Login page. Add the following to allow subsequent pages to reference the user’s session cookie on this page:
```

File: docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+  </script>
+
+
+```
+## Creating Session Cookie in FastStore Store for Localhost or Internal URLs
+The service app generates the session cookie on the main domain. Still, if the store is using localhost or an internal URL at the checkout, it is necessary to create the cookie manually in the faststore store. To do so, navigate to the checkout URL and copy the cookie value. For example:
+
+![session-cookie-example](https://user-images.githubusercontent.com/67270558/232832503-e1426219-caa2-487d-a220-1c26a8226bbf.png)
+
+Go back to the Fast Store store and use the following command to generate the cookie that will be used to sync the `orderForm`:
```
Comment (eduardoformiga) [inline docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md]: Where we might run this command? Devtools console?  Would be good explicit it :)

I noticed different patterns: `Faststore`, `Fast Store`, `FastStore`, `faststore`. WDYT about keeping the same pattern?

File: docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+```JavaScript
+document.cookie = "vtex_session={{value of session cookie}}";
+
+```
+After that, you should see the cookie created and the order form synced.
```
Comment (eduardoformiga) [inline docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md]: ⛏️ Nitpicking: the same thing related to the pattern: `order form`, `OrderForm`, `orderForm`.

**Review by eduardoformiga — COMMENTED** [review]
Good Job, @Mariana-Caetano !
I left some suggestions along with the review. I also noticed that the syntax highlight was not applied using the \```JavaScript, maybe using \```js?

File: docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md (hunk 6 above)
Comment (mariana-caetano) [reply docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md]: Good point! Just added about the console and also follow the pattern as `FastStore`.

**Review by eduardoformiga — APPROVED** [review]

File: docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+            variant="tertiary"
+            icon={<Icon name="User" width={18} height={18} weight="bold" />}
+```
+#### `src/pages/account.tsx` 
+Responsible for rendering the My Account page. Add the following to allow subsequent pages to reference the user’s session cookie on this page:
+
+```tsx
+        function Page() {
+            useEffect(() => {
+            +  window.location.href = `${storeConfig.accountUrl}${window.location.search}`
```
Comment (eduardoformiga) [inline docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md]:
```suggestion
              window.location.href = `${storeConfig.accountUrl}${window.location.search}`
```

File: docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+            return (
+```
+
+#### `src/pages/login.tsx` 
+Responsible for rendering the Login page. Add the following to allow subsequent pages to reference the user’s session cookie on this page:
+
+```tsx
+        function Page() {
+        useEffect(() => {
+        +    window.location.href = +`${storeConfig.loginUrl}${window.location.search}`
```
Comment (eduardoformiga) [inline docs/how-to-guides/platform-integration/vtex/implementing-order-form-sync.md]:
```suggestion
            window.location.href = +`${storeConfig.loginUrl}${window.location.search}`
```

**Pull request merged**